framework/src/org/checkerframework/framework/type/TypesIntoElements.java

```diff
@@ -244,7 +244,8 @@ private static void addUniqueTypeCompounds(Types types, Symbol sym, List<TypeCom
         List<Attribute.TypeCompound> res = List.nil();
 
         for (Attribute.TypeCompound tc : tcs) {
-            if (!TypeAnnotationUtils.isTypeCompoundContained(types, raw, tc)) {
+            // Don't insert an annotation if one with the same name already exists.
+            if (!TypeAnnotationUtils.isTypeCompoundContainedByName(types, raw, tc)) {
```

Comment:
If we want to write out the canonicalized version of an annotation that appears in source code, then we will have to remove the existing annotation from the element.  I don't know how to do that. Werner, is that something you could do?